### DISCUSSION TITLE
Character doesn't read player keyboard input while ragdolled in multiplayer.

### DISCUSSION THREAD

**Discussion by iridescent-lightning** [FakeFishGames/Barotrauma#11942] [Ideas and suggestions]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

Perhaps this is a wired request. Why would you want to read keyboard input in ragdoll mode since you can't move anyways? Well, because I'm using Lua to control the rope force applied on the character.   

The codes I currently use can allow players to pre-set pulling mode using "LeftShift" and "LeftCtrl". This allows players to use only one type of ammo to go up or down. However, I had a better option, it doesn't work in multiplayer since only the CLIENT can read the IsKeyDown 

Example test:
Hook.Add("InputTest", "InputTest", function(effect, deltaTime, item, targets, worldPosition)
local character = effect.user
 if character.IsKeyDown(9) then -- key code for ‘9’ is used to go down
       print('9')
        --do something
    elseif character.IsKeyDown(8) then -- key code for ‘8’ is used to go up
       print('8')
         --do something
    end
end)
--these codes only work on the CLIENT, so you won't read the server message in the console. Thus, whatever the action it does will likely also cause glitches.

### Reproduction steps

1. Install the client-side LuaForBarotrauma
2. <StatusEffect type="OnSecondaryUse" target="Character"
          <LuaHook name="InputTest"/
        </StatusEffect
Append this StatusEffect to a revolver. And also create a Lua script I posted above.
3. Start a multiplayer game.
4. Aim the revolver. Holding the ragdoll key and also press shift or ctrl.
5. Open the console and you will find that only client has received the keyboard input.

### Bug prevalence

Happens every time I play

### Version

v1.0.9.0

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report!

I wouldn't really consider this a bug report, so I think the discussions are a more appropriate place for it. Moreover, if you need a client to be able to report inputs to the server while ragdolled, I think that's something you need to implement yourself in your mod.

**Reply by iridescent-lightning** [reply]
Thanks. I will see if I can do something about it with Lua. I have fixed the grapple so it would be cool to see if you can adjust the pull force real-time.
